add warning about lack of support for insecure tls 1.2 cipher suites (#17365)
cockroachdb/docs: src/current/_includes/common/tls-bad-cipher-warning.md

{{site.data.alerts.callout_info}}

> CockroachDB clusters support TLS 1.2 or TLS 1.3 encryption and authentication for SQL clients. However, the following less-secure TLS 1.2 cipher suites are rejected by default, in accordance with the IETF's recommended cipher list defined in [RFC 8447](https://datatracker.ietf.org/doc/html/rfc8447):

Reviewer: ```suggestion
CockroachDB clusters support TLS 1.2 or TLS 1.3 encryption for SQL clients. However, the following less-secure TLS 1.2 cipher suites are rejected by default, in accordance with the IETF's recommended cipher list defined in [RFC 8447](https://datatracker.ietf.org/doc/html/rfc8447):
```

> CockroachDB clusters can talk TLS 1.2 or TLS 1.3 to SQL clients for encryption-in-flight and to prove each other's identities. However, the following less-secure TLS 1.2 cipher suites are rejected, in accordance with the IETF's recommended cipher list defined in [RFC 8447](https://datatracker.ietf.org/doc/html/rfc8447):

Reviewer 2: Perhaps I'm not up-to-speed with the security lingo, but "talk" sounds a little odd. Would switching out for "can talk" to "support" work?

> CockroachDB clusters can talk TLS 1.2 or TLS 1.3 to SQL clients for encryption-in-flight and to prove each other's identities. However, the following less-secure TLS 1.2 cipher suites are rejected, in accordance with the IETF's recommended cipher list defined in [RFC 8447](https://datatracker.ietf.org/doc/html/rfc8447):

Author: ```suggestion
CockroachDB clusters support TLS 1.2 or TLS 1.3 encryption and authentication for SQL clients. However, the following less-secure TLS 1.2 cipher suites are rejected by default, in accordance with the IETF's recommended cipher list defined in [RFC 8447](https://datatracker.ietf.org/doc/html/rfc8447):
```

- `TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA`
- `TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA`
- `TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA`
- `TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA`
- `TLS_RSA_WITH_AES_128_GCM_SHA256`
- `TLS_RSA_WITH_AES_256_GCM_SHA384`
- `TLS_RSA_WITH_AES_128_CBC_SHA`
- `TLS_RSA_WITH_AES_256_CBC_SHA`

> SQL clients or intermediate proxies or load balancers that do not support any cipher suites that CockroachDB considers to be secure and supports will be unable to connect to CockroachDB clusters, unless they are configured to allow them. To allow SQL connections using the deprecated cipher suites, set the `COCKROACH_TLS_ENABLE_OLD_CIPHER_SUITES` environment variable to `true` for your `cockroach start` command. This mode is not recommended, unless you must use a client, intermediate proxy, or load balancer that doesn't support any of the more secure cipher suites.

Reviewer 2: Oh that is a lot of information to get across. And tough to write. Might just be me, but I had to read it a few times to fully understand what is not supporting what, or won't work.

What about making the first sentence a list and breaking apart the sentence a little bit, something like the following:

> SQL clients, intermediate proxies, or load balancers that do not support any cipher suites that CockroachDB considers to be secure and supports will be unable to connect to CockroachDB clusters. Unless, you configure them to allow the cipher suites.

That ^ might not actually be correct, the second sentence here. I couldn't tell from the original what "they" referred to for sure.

Or, 

> You must configure SQL clients, intermediate proxies, or load balancers to allow cipher suites that are unsupported, which CockroachDB considers to be secure and supports. Otherwise your client, proxy, or load balancer will not be able to connect to a CockroachDB cluster.
> 
> To allow the deprecated cipher suites....

Author: hopefully that simplifies a little

> SQL clients or intermediate proxies or load balancers that do not support any cipher suites that CockroachDB considers to be secure and supports will be unable to connect to CockroachDB clusters, unless they are configured to allow them. To allow SQL connections using the deprecated cipher suites, set the `COCKROACH_TLS_ENABLE_OLD_CIPHER_SUITES` environment variable to `true` for your `cockroach start` command. This mode is not recommended, unless you must use a client, intermediate proxy, or load balancer that doesn't support any of the more secure cipher suites.

Author: ```suggestion
SQL clients, intermediate proxies, or load balancers that do not support any cipher suites that CockroachDB supports will be unable to connect to CockroachDB clusters. To allow SQL connections using the deprecated cipher suites, set the `COCKROACH_TLS_ENABLE_OLD_CIPHER_SUITES` environment variable to `true` for your `cockroach start` command. This mode is not recommended, unless you must use a client, intermediate proxy, or load balancer that doesn't support any of the more secure cipher suites.
```

{{site.data.alerts.callout_info}}

> CockroachDB clusters accept TLS 1.2 or TLS 1.3 security certificates for client authentication and encryption-in-flight. However, certain less-secure TLS 1.2 cipher suites are rejected, in accordance with the IETF's recommended cipher list defined in [RFC 8447](https://datatracker.ietf.org/doc/html/rfc8447).

Reviewer: ```suggestion
CockroachDB clusters can talk TLS 1.2 or TLS 1.3 to SQL clients for encryption-in-flight and to prove each other's identities. However, certain less-secure TLS 1.2 cipher suites are rejected, in accordance with the IETF's recommended cipher list defined in [RFC 8447](https://datatracker.ietf.org/doc/html/rfc8447).
```

> SQL clients that are more than five years old may fail to connect, if they do not support any cipher suites that CockroachDB considers to be secure, and hence supports.

Reviewer: ```suggestion
SQL clients or intermediate proxies or load balancers that do not support any cipher suites that CockroachDB considers to be secure and supports, may fail to connect to a cluster.
```

> To allow connections using these deprecated cipher suites, and hence support these older clients, you can set the `COCKROACH_TLS_ENABLE_OLD_CIPHER_SUITES` environment variable to `true` for your `cockroach start` command. However, this is not recommended as these cipher suites are considered insecure.

Reviewer: ```suggestion
To allow SQL connections using the deprecated cipher suites, you can set the `COCKROACH_TLS_ENABLE_OLD_CIPHER_SUITES` environment variable to `true` for your `cockroach start` command. We provide the above environment variable only for cases when you can't use a client or intermediate proxy or load balancer that support any of the secure cipher suites.
```

> Refer to the code for the current list of supported cipher suites: [CockroachDB on GitHub: tls_ciphersuites.go](https://github.com/cockroachdb/cockroach/blob/master/pkg/security/tls_ciphersuites.go)

Reviewer: Hmm.. I'm not sure if we should link to a code file in Github repo. Is there a precedent for it across the docs?

Author: I don't see why not, the code is open source. If the truth changes, it'll be recorded here, so this is the canonical reference. I'll ping the team for a consensus.

Author: Ok, removed the link and just have the list now.
